mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2057,14 +2058,41 @@ where
         },
     };
 
+    let mut updated = false;
     let mut history_map: HashMap<H256Json, TransactionDetails> = history
         .into_iter()
-        .filter_map(|tx| {
+        .filter_map(|mut tx| {
             let tx_hash = H256Json::from_str(&tx.tx_hash).ok()?;
-            Some((tx_hash, tx))
+
+            // Quick fix for null valued coin fields in fee details of old tx history entries
+            match tx.fee_details {
+                Some(Utxo(ref mut fee_details)) => {
+                    if fee_details.coin.is_none() {
+                        fee_details.coin = Some(String::from(&tx.coin));
+                        updated = true;
+                    }
+                    Some((tx_hash, tx))
+                },
+                Some(_) => None,
+                None => Some((tx_hash, tx)),
+            }
         })
         .collect();
 
+    if updated {
```

Comment:
`process_history_loop` is already too big 🙂 Could you create a separate `migration` file and check its contents? If it doesn't exist or has a value below 2, we should migrate the history database, setting coins tickers. This way it will be done only once. When loop runs next time, it will be aware that migration was done, so no additional history loading and reiteration will be required. Please also move this migration code to a separate fn.

Reply:
> Could you create a separate `migration` file and check its contents? If it doesn't exist or has a value below 2, we should migrate the history database, setting coins tickers.

Do you mean a file like this: 

```
KMD=2
RICK=1
```

Reply:
@caglaryucekaya Tx history file name is `COIN_ADDRESS.json`, so for migration it can be `COIN_ADDRESS_migration`. It will have only migration number as content, e.g. `1`.

Reply:
Okay thanks. I wasn't sure if you were thinking a single migration file for all coins, or a separate one for each.